rpc/handler.go

```diff
@@ -479,33 +511,9 @@ func (h *handler) runMethod(ctx context.Context, msg *jsonrpcMessage, callb *cal
 	stream.WriteObjectField("result")
 	_, err := callb.call(ctx, msg.Method, args, stream)
 	if err != nil {
-		//return msg.errorResponse(err)
-
-		stream.WriteMore()
-		stream.WriteObjectField("error")
-		stream.WriteObjectStart()
-		stream.WriteObjectField("code")
-		ec, ok := err.(Error)
-		if ok {
-			stream.WriteInt(ec.ErrorCode())
-		} else {
-			stream.WriteInt(defaultErrorCode)
-		}
-		stream.WriteMore()
-		stream.WriteObjectField("message")
-		stream.WriteString(fmt.Sprintf("%v", err))
-		de, ok := err.(DataError)
-		if ok {
-			stream.WriteMore()
-			stream.WriteObjectField("data")
-			data, derr := json.Marshal(de.ErrorData())
-			if derr == nil {
-				stream.Write(data)
-			} else {
-				stream.WriteString(fmt.Sprintf("%v", derr))
-			}
-		}
-		stream.WriteObjectEnd()
+		stream.WriteNil();
```

Comment:
I think this “nil” makes json invalid. There was no nil and no WriteMore before.

Reply:
No, because now the nil is setting the result field to "null" which is valid. Then WriteMore inserts the comma before more object fields are added from handle error (error and message). I did take a moment to test the error case where fromBlock > toBlock and the result is valid JSON and what I would expect.

Reply:
When any of those errors are returned, the result field is always be uninitialized, so setting it to null or an empty array are the 2 options I saw.

Reply:
make sense. then good to go. let me test some other methods